_posts/2019-05-10-why-projects-need-codenames.markdown

```diff
@@ -0,0 +1,48 @@
+---
+layout: epic
+title: Why Projects Need Code Names
+date: 2019-05-10
+author: [joey]
+categories: [engineering, git, github]
+---
+
+Before I joined Artsy, I worked at companies where software projects tended to have meaningful, predictable names. If we were building a system for flagging media uploads, it might be called `media-review`. In many cases, our code repositories' names matched the main product's branding or even the company's name. Life was simple and there was no risk of ambiguity.
+
+At Artsy, our systems have peculiar code names like _Gravity_, _Pulse_, and _Vortex_. There's a persistent learning curve as you contribute to different repositories or as new services get created. Numerous times, I've wondered: are code names worth the trouble?
+
+<!-- more -->
+
+![](/images/2019-05-10-why-projects-need-codenames/github_projects.png)
+
+To be clear, _any_ project naming scheme works in small quantities. Personal projects or libraries for public release should probably just be named for clarity. Code names start to offer benefits as a team grows and a product evolves over the longer term:
```

Comment:
> Personal projects or libraries for public release should probably just be named for clarity.

This could do with some elaboration, I’m not sure I agree that this _should_ probably be better.